shell/platform/android/io/flutter/embedding/engine/FlutterEngine.java

```diff
@@ -374,6 +376,7 @@ public void destroy() {
     platformViewsController.onDetachedFromJNI();
     dartExecutor.onDetachedFromJNI();
     flutterJNI.removeEngineLifecycleListener(engineLifecycleListener);
+    flutterJNI.setDynamicFeatureManager(null);
     flutterJNI.detachFromNativeAndReleaseResources();
```

Comment:
if FlutterInjector.instance().dynamicFeatureManager() is not null, call destroy

Reply:
You didn't address this one. I think nothing's calling DynamicFeatureManager.destroy

Reply:
Github UI didn't register this one changing as well, see lines below.